src/Open3D/GUI/NumberEdit.cpp

```diff
@@ -0,0 +1,187 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "NumberEdit.h"
+
+#include "Theme.h"
+#include "Util.h"
+
+#include <imgui.h>
+
+#include <algorithm>  // for min, max
+#include <cmath>
+#include <sstream>
+#include <unordered_set>
+
+#include <string.h>
+
+namespace open3d {
+namespace gui {
+
+namespace {
+static int gNextNumberEditId = 1;
+}
+
+struct NumberEdit::Impl {
+    std::string id;
+    NumberEdit::Type type;
+    // Double has 24-bits of integer range, which is sufficient for the
```

Comment:
"24-bits" -> "53-bits"  (24-bits is float)

Reply:
Oops, you're right. Changed